internal/provider/resources/countrycodeallowlist.go

```diff
@@ -42,6 +43,36 @@ type countryCodeAllowlistModel struct {
 	LastUpdated    types.String `tfsdk:"last_updated"`
 }
 
+func standardizedCountryCodes(countryCodes []string) []string {
+	// Standardize country codes to uppercase and remove duplicates.
+	standardizedCodesSet := map[string]bool{}
+	for _, countryCode := range countryCodes {
+		countryCode = strings.ToUpper(countryCode)
+		standardizedCodesSet[countryCode] = true
+	}
+	standardizedCodes := make([]string, 0, len(standardizedCodesSet))
+	for countryCode := range standardizedCodesSet {
+		standardizedCodes = append(standardizedCodes, countryCode)
+	}
+	// Sort the standardized country codes for consistency.
+	sort.Strings(standardizedCodes)
+	return standardizedCodes
+}
+
+func areCountryCodesEqual(a, b []string) bool {
```

Comment:
[dust] Tiniest of wording nits
```suggestion
func areCountryCodesEquivalent(a, b []string) bool {
```